File: python/e20.branch.py
#!/usr/bin/python
import sys  # system library for standard input and output

# build a menu string to output to user
menu = """\
Select an item from the menu.

  A - Coffee
  B - Espresso
  C - Latte
  D - Machiato
  E - Capucino
  F - Mocha
  G - Tea

Make your selection: """
 
# get user output 
sys.stdout.write(menu)         # output prompt
selection = sys.stdin.read(1)  # read one character
# convert to uppercase if user entered lowercase
selection = selection.upper()            

# test on a single variable selection
if selection == "A":
  print "You selected a Coffee"
elif selection == "B":
  print "You selected an Espresso"
elif selection == "C":
  print "You selected a Latte"
elif selection == "D":
  print "You selected a Machiato"
elif selection == "E":
  print "You selected a Capucino"
elif selection == "F":
  print "You select a Mocha"
elif selection == "G":
  print "You selected a Tea"
else:
  print "You have not entered a valid selection"

File: wsh.jscript/e20.branch.js
// prompt and read one character
WScript.stdout.write("Input a character: ");
keypress = WScript.stdin.read(1);

// find RegExp and print result
if (RegExp("[a-z]").test(keypress)) 
    WScript.Echo("Lowercase letter");
else if (RegExp("[A-Z]").test(keypress)) 
   WScript.Echo("Uppercase letter");
else if (RegExp("[0-9]").test(keypress)) 
    WScript.Echo("Digit");
else 
    WScript.Echo("Punctuation, whitespace, or other");
